packages/lab/src/badge/Badge.tsx

```diff
@@ -60,32 +34,40 @@ export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
     window: targetWindow,
   });
 
-  const badgeId = useId();
-  const childId = useId(
-    isValidElement<HTMLAttributes<HTMLElement>>(children)
-      ? children.props?.id
-      : undefined
-  );
+  let valueText = value;
 
-  let badgeContentValue = badgeContent;
-  if (badgeContentValue > max) {
-    badgeContentValue = `${max}+`;
+  if (typeof value === "number" && !max) {
+    valueText = value > 999 ? "999+" : value;
+  } else if (typeof value === "number" && max) {
+    valueText = value > max ? `${max}+` : value;
+  } else if (typeof value === "string") {
+    valueText = value.length > 4 ? `${value.slice(0, 4)}...` : value;
   }
-
   return (
-    <span
-      className={clsx(withBaseName(), className)}
-      ref={ref}
-      role="img"
-      aria-labelledby={clsx(childId, badgeId)}
-      {...rest}
-    >
-      {children && isValidElement<HTMLAttributes<HTMLElement>>(children)
-        ? cloneElement(children, { id: childId })
-        : children}
-      <span id={badgeId} className={clsx(withBaseName("badge"))}>
-        {badgeContentValue}
-      </span>
+    <span className={clsx(withBaseName(), className)} ref={ref} {...rest}>
+      {children}
+      {typeof value === "string" && value.length > 4 ? (
+        <>
+          <span
+            className={clsx(withBaseName("badge"), {
+              [withBaseName("topRight")]: children,
+            })}
+            aria-hidden="true"
+          >
+            {valueText}
+          </span>
+          <span className="salt-visuallyHidden">{valueText}</span>
+        </>
+      ) : (
+        <span
+          className={clsx(withBaseName("badge"), {
+            [withBaseName("topRight")]: children,
+          })}
+          aria-hidden="true"
```

Comment:
Remove this aria-hidden

Reply:
In the case where there is no truncation we want to render value as is and not hide anything

Reply:
yep - done